### DISCUSSION TITLE
Auto update CodeQL database

### DISCUSSION THREAD

**Discussion by doublevkay** [community/community#23946] [New to GitHub]
<p>Hi, I’m trying to learn how to write my own CodeQL queries. But any time I change my source code, I need to create a new database <code>codeql database create</code>.<br>
Is there any solution to auto update codeql database?</p>

**Comment by criemen** [comment, marked as answer]
<p>Hi <a class="mention" href="/vovikhangcdv">@vovikhangcdv</a> ,<br>
great that you’re learning CodeQL!</p>
<p>Unfortunately, there is no solution to auto-update the CodeQL database.<br>
There is a CodeQL action, so that you can get a database (and run the analysis) as part of a CI setup, but I don’t think that that’s very useful if you’re iterating (mostly) on the queries.<br>
Maybe you can write a shell script that contains all the relevant commands?</p>
